controllers/helmrepository_controller_test.go

```diff
@@ -1299,3 +1300,75 @@ func TestHelmRepositoryReconciler_ReconcileSpecUpdatePredicateFilter(t *testing.
 		return false
 	}, timeout).Should(BeTrue())
 }
+
+func TestHelmRepositoryReconciler_InMemoryCaching(t *testing.T) {
+	g := NewWithT(t)
+	testCache.Clear()
+
+	testServer, err := helmtestserver.NewTempHelmServer()
+	g.Expect(err).NotTo(HaveOccurred())
+	defer os.RemoveAll(testServer.Root())
+
+	g.Expect(testServer.PackageChartWithVersion("testdata/charts/helmchart", "0.1.0")).To(Succeed())
+	g.Expect(testServer.GenerateIndex()).To(Succeed())
+
+	testServer.Start()
+	defer testServer.Stop()
+
+	ns, err := testEnv.CreateNamespace(ctx, "helmrepository")
+	g.Expect(err).ToNot(HaveOccurred())
+	defer func() { g.Expect(testEnv.Delete(ctx, ns)).To(Succeed()) }()
+
+	helmRepo := &sourcev1.HelmRepository{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "helmrepository-",
+			Namespace:    ns.Name,
+		},
+		Spec: sourcev1.HelmRepositorySpec{
+			URL: testServer.URL(),
+		},
+	}
+	g.Expect(testEnv.CreateAndWait(ctx, helmRepo)).To(Succeed())
+
+	key := client.ObjectKey{Name: helmRepo.Name, Namespace: helmRepo.Namespace}
+	// Wait for finalizer to be set
+	g.Eventually(func() bool {
+		if err := testEnv.Get(ctx, key, helmRepo); err != nil {
+			return false
+		}
+		return len(helmRepo.Finalizers) > 0
+	}, timeout).Should(BeTrue())
+
+	r := HelmRepositoryReconciler{
+		Client:        testEnv,
+		EventRecorder: record.NewFakeRecorder(32),
+		Metrics:       testMetricsH,
+		Getters:       testGetters,
+		Storage:       testStorage,
+		Cache:         testCache,
+		TTL:           1 * time.Second,
+	}
+
+	if err := (&r).SetupWithManager(testEnv); err != nil {
+		panic(fmt.Sprintf("Failed to start HelmRepositoryReconciler: %v", err))
+	}
```

Comment:
we don't need this. It is setup in suite_test.go

Reply:
removed